bindings/postgres/postgres.go

```diff
@@ -73,11 +73,20 @@ func (p *Postgres) Init(ctx context.Context, meta bindings.Metadata) error {
 
 	// This context doesn't control the lifetime of the connection pool, and is
 	// only scoped to postgres creating resources at init.
-	p.db, err = pgxpool.NewWithConfig(ctx, poolConfig)
+	connCtx, connCancel := context.WithTimeout(ctx, m.Timeout)
+	p.db, err = pgxpool.NewWithConfig(connCtx, poolConfig)
+	connCancel()
```

Comment:
Was this meant to be deferred? Wouldn't we still need the `p.db` connection for the ping below?

Reply:
Ha! Josh was faster :D

Reply:
we have this exact same code in the postgres state store